src/charm.py

```diff
@@ -404,14 +418,10 @@ def _unit_ip(self) -> str:
     def is_external_client(self) -> Optional[str]:
         """Returns the connectivity mode which mongos should use.
 
-        This is determined by checking the modes requested by the client(s).
-
-        TODO: Future PR. This should be modified to work for many clients.
+        Note that for K8s routers this should always default to True. However we still include
+        this function so that we can have parity on properties with the K8s and VM routers.
         """
-        if EXTERNAL_CONNECTIVITY_TAG not in self.app_peer_data:
-            return False
-
-        return json.loads(self.app_peer_data.get(EXTERNAL_CONNECTIVITY_TAG))
+        return True
```

Comment:
**suggestion:** Could you do something like:

```python
return any([relation.get("external-node-connectivity", "") for relation in self.model.relations["mongodb_client"]])

Reply:
I only mention because it's not clear what `is_external_client` means in this context. I think you mean `is_k8s`, right?

I think down the line when you have clients, and you need to decide what to set `uris` to, you'd have something different, like:
```python
uris = f"{node_ip}:{node_port}" if client.external_node_connectivity else f"{dns_name}:{normal_port}"
```

Reply:
Hey! I like where your head is at here. This will have to be addressed later on in a PR that handles external-client relations.

I say this because

1. `is_external_client` is used in the shared lib `ls ~/mongos-k8s-operator/lib/charms/mongodb/v0/config_server_interface.py` and if we want to modify the functionality of this function we will need to update that lib (which is owned by MongoDB VM). 
2. the update use of `external_client` is a paradigm shift from the original so its not quite as simple as what you have described above. If you'd like to chat about it lmk :)

Reply:
Does VM have a concept of 'external'? What does it mean in that context?
I looked at the lib you mentioned, was I misunderstanding what 'external' means here?
Am I correct in thinking that 'external' really means 'not a socket address, but an IP', external to the unit, not K8s?

Reply:
> Does VM have a concept of 'external'? What does it mean in that context?

Yes, it means that  'external' means 'not a socket address, but an IP'

But further:

VM router only supports **one client** . What does this mean:
1.  we use `is_external` for deciding if we start with a a socket address vs IP
2. we use `is_external` to determine what to supply the **one integrated client** with (IP/socket)

K8s router supports **multiple clients**. Additionally
1. we always start as external
2. we support multiple clients, so we must have a function called `is_client_external` which returns False if the client should use K8s DNS or True if the client should use NodePort

In the shared lib for managing these connections is currently in the VM-paradigm, so we will need to update it for the K8s paradigm

Reply:
All makes sense! Personally, I think I would use `is_local` for a socket address bool check, and `is_external` for an 'outside k8s' bool check, but it's your call, no issue.